components/net/fetch/methods.rs

```diff
@@ -271,22 +271,43 @@ pub fn main_fetch(request: Rc<Request>,
 
     let mut response_loaded = false;
     {
-        // Step 14
+        let invalid_mime = {
+            let mime_type = response.headers.get::<ContentType>();
+            let csv: Mime = "text/csv".parse().unwrap();
+            mime_type.is_some() && (match *mime_type.unwrap() {
+                ContentType(Mime(TopLevel::Audio, _, _)) |
+                ContentType(Mime(TopLevel::Video, _, _)) |
+                ContentType(Mime(TopLevel::Image, _, _)) => true,
+                ContentType(ref m_type) => { *m_type == csv }
+            })
+        };
+
         let network_error_res;
-        let internal_response = if let Some(error) = response.get_network_error() {
+        let mut step16 = false;
+        let internal_response = if invalid_mime && request.type_ == Type::Script {
```

Comment:
No, because `internal_response` is a reference to `response`, they can't be changed at the same time due to issues with borrowing. So, `internal_response` had to be set at once as a result of step 14 and step 16.